src/PrestaShopBundle/EventListener/DemoModeEnabledListener.php

```diff
@@ -165,4 +172,25 @@ private function getAnnotation($controllerObject, $methodName)
 
         return $this->annotationReader->getMethodAnnotation($reflectionMethod, $tokenAnnotation);
     }
+
+    /**
+     * Gets query parameters by comparing them to the current request attributes.
+     *
+     * @param array $queryParametersToKeep
+     * @param Request $request
+     *
+     * @return array
+     */
+    private function getQueryParamsFromRequestQuery(array $queryParametersToKeep, Request $request)
+    {
+        $result = [];
+
+        foreach ($queryParametersToKeep as $queryParameterName) {
+            if ($value = $request->get($queryParameterName)) {
```

Comment:
Please do not assign variables inside `if` statement. It works, but it is not easy to read, and it might have unwanted consequences. For example what if `$request->get($queryParameterName)` returns `0` ? Then it is going to be skipped.

Reply:
done - thank you! :+1: